gusto/initialisation_tools.py

```diff
@@ -0,0 +1,94 @@
+"""
+A module containing some tools for computing initial conditions, such
+as balanced initial conditions.
+"""
+
+from __future__ import absolute_import
+from firedrake import MixedFunctionSpace, TrialFunctions, TestFunctions, \
+    FacetNormal, inner, div, dx, ds_b, ds_t, DirichletBC, \
+    Expression, Function, Constant, \
+    LinearVariationalProblem, LinearVariationalSolver
+
+
+def compressible_hydrostatic_balance(state, theta0, rho0,
+                                     top=True, pi_boundary=Constant(1.0),
```

Comment:
Oh I see. Yes, you can set the defaults to your test case if you like.

cheers
--cjc

On 1 July 2016 at 17:40, jshipton <notifications@github.com> wrote:

> In gusto/initialisation_tools.py
> <https://github.com/firedrakeproject/gusto/pull/48#discussion_r69326173>:
>
> > @@ -0,0 +1,94 @@
> > +"""
> > +A module containing some tools for computing initial conditions, such
> > +as balanced initial conditions.
> > +"""
> > +
> > +from __future__ import absolute_import
> > +from firedrake import MixedFunctionSpace, TrialFunctions, TestFunctions, \
> > +    FacetNormal, inner, div, dx, ds_b, ds_t, DirichletBC, \
> > +    Expression, Function, Constant, \
> > +    LinearVariationalProblem, LinearVariationalSolver
> > +
> > +
> > +def compressible_hydrostatic_balance(state, theta0, rho0,
> > +                                     top=True, pi_boundary=Constant(1.0),
>
> Of course - but I notice you're not using the defaults so I wondered why
> they'd been chosen like this.
> ________________________________ From: Colin J Cotter <
> notifications@github.com> Sent: 01 July 2016 17:36:25 To:
> firedrakeproject/gusto Cc: Shipton, Jemma; Comment Subject: Re:
> [firedrakeproject/gusto] Hydrostatic initialisation and use of geopotential
> (#48 <https://github.com/firedrakeproject/gusto/pull/48>) In
> gusto/initialisation_tools.py<#48 (comment)
> <https://github.com/firedrakeproject/gusto/pull/48#discussion_r69325776>>:
> @@ -0,0 +1,94 @@ +""" +A module containing some tools for computing
> initial conditions, such +as balanced initial conditions. +""" + +from
> __future__ import absolute_import +from firedrake import
> MixedFunctionSpace, TrialFunctions, TestFunctions, \ + FacetNormal, inner,
> div, dx, ds_b, ds_t, DirichletBC, \ + Expression, Function, Constant, \ +
> LinearVariationalProblem, LinearVariationalSolver + + +def
> compressible_hydrostatic_balance(state, theta0, rho0, + top=True,
> pi_boundary=Constant(1.0),
> It depends on the test problem. ... On 1 July 2016 at 17:24, jshipton
> ***@***.***> wrote: In gusto/initialisation_tools.py <#48
> <https://github.com/firedrakeproject/gusto/pull/48> (comment)<#48
> (comment)
> <https://github.com/firedrakeproject/gusto/pull/48#discussion_r69324335>>>:
> > @@ -0,0 +1,94 @@ > +""" > +A module containing some tools for computing
> initial conditions, such > +as balanced initial conditions. > +""" > + >
> +from __future__ import absolute_import > +from firedrake import
> MixedFunctionSpace, TrialFunctions, TestFunctions, \ > + FacetNormal,
> inner, div, dx, ds_b, ds_t, DirichletBC, \ > + Expression, Function,
> Constant, \ > + LinearVariationalProblem, LinearVariationalSolver > + > + >
> +def compressible_hydrostatic_balance(state, theta0, rho0, > + top=True,
> pi_boundary=Constant(1.0), I'm not sure what the default is for the sphere,
> but using this for the slice I would want the default to be to set pi=1. on
> the bottom i.e. top=False and pi_boundary=Constant(1.0). Do these current
> defaults make sense on the sphere? - You are receiving this because you
> authored the thread. Reply to this email directly, view it on GitHub <
> https://github.com/firedrakeproject/gusto/pull/48/files/058df01ad2c507ed69be78644852c2e74f1a8d18#r69324335>,
> or mute the thread <
> https://github.com/notifications/unsubscribe/AF0q8jaybuK8pLFlICNwCTdKnPwBqYAXks5qRT83gaJpZM4JC5M5>
> . -- http://www.imperial.ac.uk/people/colin.cotter
> www.cambridge.org/9781107663916<http://www.cambridge.org/9781107663916
> <http://www.cambridge.org/9781107663916%3Chttp://www.cambridge.org/9781107663916>>
> - You are receiving this because you commented. Reply to this email
> directly, view it on GitHub<
> https://github.com/firedrakeproject/gusto/pull/48/files/058df01ad2c507ed69be78644852c2e74f1a8d18#r69325776>,
> or mute the thread<
> https://github.com/notifications/unsubscribe/AGjpyga96mpaPAw7i0QPEOxSf-_wo8Mmks5qRUIJgaJpZM4JC5M5
> >.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/firedrakeproject/gusto/pull/48/files/058df01ad2c507ed69be78644852c2e74f1a8d18#r69326173>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AF0q8h5xXrluxoHrmkvDxQaItn9roA-sks5qRULlgaJpZM4JC5M5>
> .
>



-- 
http://www.imperial.ac.uk/people/colin.cotter

www.cambridge.org/9781107663916